src/meili/match_route.cc

```diff
@@ -218,7 +218,7 @@ bool MergeRoute(const State& source, const State& target, std::vector<EdgeSegmen
 
   // MergeEdgeSegments(route, segments.rbegin(), segments.rend());
   route.insert(route.end(), segments.crbegin(), segments.crend());
-  return true;
+  return !segments.empty();
```

Comment:
@yuzheyan i think this contradicts the other change i made on friday. im adding a TODO to the code to fix it upstream in routing.cc